node-commons/src/test/java/org/apache/iotdb/commons/path/PartialPathTest.java

```diff
@@ -23,8 +23,496 @@
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
 public class PartialPathTest {
 
+  @Test
+  public void testLegalPath() throws IllegalPathException {
+    String[] nodes;
+    // empty path
+    PartialPath a = new PartialPath("", false);
+    Assert.assertEquals("", a.getFullPath());
+    Assert.assertEquals(0, a.getNodes().length);
+
+    // suffix path
+    PartialPath b = new PartialPath("s1");
+    Assert.assertEquals("s1", b.getFullPath());
+    Assert.assertEquals("s1", b.getNodes()[0]);
+
+    // normal node
+    PartialPath c = new PartialPath("root.sg.a");
+    Assert.assertEquals("root.sg.a", c.getFullPath());
+    nodes = new String[] {"root", "sg", "a"};
+    checkNodes(nodes, c.getNodes());
+
+    // quoted node
+    PartialPath d = new PartialPath("root.sg.`a.b`");
+    Assert.assertEquals("root.sg.`a.b`", d.getFullPath());
+    nodes = new String[] {"root", "sg", "`a.b`"};
+    checkNodes(nodes, d.getNodes());
+
+    PartialPath e = new PartialPath("root.sg.`a.``b`");
+    Assert.assertEquals("root.sg.`a.``b`", e.getFullPath());
+    nodes = new String[] {"root", "sg", "`a.``b`"};
+    checkNodes(nodes, e.getNodes());
+
+    PartialPath f = new PartialPath("root.`sg\"`.`a.``b`");
+    Assert.assertEquals("root.`sg\"`.`a.``b`", f.getFullPath());
+    nodes = new String[] {"root", "`sg\"`", "`a.``b`"};
+    checkNodes(nodes, f.getNodes());
+
+    PartialPath g = new PartialPath("root.sg.`a.b\\\\`");
+    Assert.assertEquals("root.sg.`a.b\\\\`", g.getFullPath());
+    nodes = new String[] {"root", "sg", "`a.b\\\\`"};
+    checkNodes(nodes, g.getNodes());
+
+    // quoted node of digits
+    PartialPath h = new PartialPath("root.sg.`111`");
+    Assert.assertEquals("root.sg.`111`", h.getFullPath());
+    nodes = new String[] {"root", "sg", "`111`"};
+    checkNodes(nodes, h.getNodes());
+
+    // quoted node of key word
+    PartialPath i = new PartialPath("root.sg.`select`");
+    Assert.assertEquals("root.sg.`select`", i.getFullPath());
+    nodes = new String[] {"root", "sg", "`select`"};
```

Comment:
It seems that this behavior is not as our expected.
IMO, 
```
root.sg.abc
```
and
```
root.sg.`abc`
```
should be same timeseries, which is actually stored as `String[]{"root", "sg", "abc"}`.

If a user wants to save backquote, or other special chars, he should use 

```
root.sg.`ab``.c`
```

and the actual timeseries will be
```
String[]{"root", "sg", "ab`.c"}
```

@qiaojialin @SteveYurongSu What do you think?

Reply:
+1 for this case

Reply:
root.sg.`abc` will be String[]{"root", "sg", "abc"}. 
root.sg.`select` will be String[]{"root", "sg", "select"}.
root.sg.`111` will be String[]{"root", "sg", "`111`"}.
root.sg.`ab``.c` will be String[]{"root", "sg", "`ab``.c`"}